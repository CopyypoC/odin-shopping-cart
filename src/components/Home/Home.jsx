import styles from "./Home.module.css";

export function Home() {
  return (
    <div className={styles.homeContainer}>
      <img
        src="/hero.jpg"
        alt="Two fashionable purses"
        className={styles.hero}
      />
      <p className={styles.heroText}>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptas, in
        ab. Nostrum atque harum ducimus. Repudiandae, voluptate omnis. Possimus
        vero eveniet voluptates? Fugiat, debitis voluptate! Ut excepturi sunt
        mollitia sapiente eum. Sunt modi blanditiis delectus aliquam illum,
        voluptates totam iure nulla, temporibus quae maxime repudiandae, saepe
        quam? Assumenda, laboriosam voluptates. Soluta iure qui eos officia
        officiis corrupti facere, in suscipit eum eaque dolorem quas cum odit.
        Cumque doloremque quo, ea unde perspiciatis dignissimos placeat officia
        harum amet at ex accusantium!
      </p>
    </div>
  );
}
