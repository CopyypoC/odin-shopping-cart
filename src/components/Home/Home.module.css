.homeContainer {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 0 auto;
  width: 70vw;
}

.hero {
  margin-top: 3rem;
  width: 70vw;
  max-height: 70vh;
  object-fit: cover;
  box-shadow: var(--box-shadow);
  border-radius: 4px;
}

.heroText {
  margin: 2rem 0;
  font-size: 1.5rem;
  font-weight: 500;
  color: var(--font-clr);
}
