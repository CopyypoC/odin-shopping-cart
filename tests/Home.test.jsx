import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import { Home } from "../src/components/Home/Home.jsx";

describe("Home component", () => {
  it("matches snapshot", () => {
    const home = render(<Home />);

    expect(home).toMatchSnapshot();
  });
});
