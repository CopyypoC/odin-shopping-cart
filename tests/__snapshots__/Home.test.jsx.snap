// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home component > matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <img
          alt="Two fashionable purses"
          src="/public/hero.jpg"
        />
        <p>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Expedita fugiat aliquid nesciunt culpa qui optio quis voluptatibus eveniet iure ut vitae, quo earum blanditiis unde adipisci. Doloremque, fugiat libero veniam consequuntur ut consequatur velit. Aliquam, magnam maiores velit incidunt, minima ratione magni quas ipsam eaque consequatur cumque laboriosam perferendis nulla!
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <img
        alt="Two fashionable purses"
        src="/public/hero.jpg"
      />
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Expedita fugiat aliquid nesciunt culpa qui optio quis voluptatibus eveniet iure ut vitae, quo earum blanditiis unde adipisci. Doloremque, fugiat libero veniam consequuntur ut consequatur velit. Aliquam, magnam maiores velit incidunt, minima ratione magni quas ipsam eaque consequatur cumque laboriosam perferendis nulla!
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
